drv/lpc55-sha256/src/lib.rs

```diff
@@ -0,0 +1,304 @@
+        //   with 448 zeros.
+        // - Add the length of the original data, _in bits,_ as a 64-bit
+        //   big-endian integer to round out the final block to 512 bits.
+        // - Add the final block(s) to the digest.
+        //
+        // (If you're curious, this construction provides a defense against
+        // messages of slightly different lengths hashing to the same value.)
+        //
+        // Since we move data in 32-bit words only, the padding process is
+        // slightly simplified here:
+        //
+        // - Append a word with only its MSB set.
+        // - Append words of zeros until two words of space remain in the block.
+        //   This may require starting a new block.
+        // - Append the high word of the data length in bits, and the low word,
+        //   in that order, as big-endian integers.
+
+        let word_count_before_padding = self.word_count;
+
+        // We want the PAD bit to be in the MSB of the first byte added to the
+        // digest, which, due to us being little-endian, means our pad value is:
+        const PAD: u32 = 0x80_00_00_00_u32.swap_bytes();
+
+        self.load_word(PAD, 0);
+        // Extend with zeros until we're aligned properly for the final length.
+        while self.word_count % WORDS_PER_BLOCK64
+            != WORDS_PER_BLOCK64 - Wrapping(2)
+        {
+            self.load_word(0, 0);
+        }
+        // We are now 14 words into a block, no synchronization is necessary, we
+        // just need to load the length of the pre-padded data in bits. As with
+        // PAD above, since these aren't round-tripping through little-endian
+        // memory, we wind up having to swap their bytes:
+        let Wrapping(length) = word_count_before_padding * Wrapping(32);
+        self.load_word(u32::swap_bytes((length >> 32) as u32), 0);
+        self.load_word(u32::swap_bytes(length as u32), 0);
+
+        // Wait for our result!
+        while self.engine.status.read().digest().is_not_ready() {
+            if self.notification_mask != 0 {
+                // Permit the hardware to generate an IRQ on DIGEST
+                self.engine.intenset.write(|w| w.digest().set_bit());
+
+                // Wait for it!
+                sys_irq_control(self.notification_mask, true);
+                let _ = sys_recv_closed(
+                    &mut [],
+                    self.notification_mask,
+                    TaskId::KERNEL,
+                );
+
+                // Turn it back off lest it spam us in the future.
+                self.engine.intenclr.write(|w| w.digest().set_bit());
+            }
+        }
+
+        // The result arrives in registers called digest0..digest7, which the
+        // PAC calls digest0[0] .. digest0[7] for some reason.
+        let mut result = [0; WORDS_PER_HASH];
+        for (i, dest) in result.iter_mut().enumerate() {
+            *dest = self.engine.digest0[i].read().bits().swap_bytes();
+        }
```

Comment:
Optional: you could remove explicit indexing with
```suggestion
        for (dest, src) in result.iter_mut().zip(&self.engine.digest0) {
            *dest = src.read().bits().swap_bytes();
        }
```